### ISSUE TITLE
Starlight range hax

### ISSUE THREAD

**Issue by GoogleFrog** [ZeroK-RTS/Zero-K#1583]
I had a battle where starlight could shoot further than its range ring suggests. http://zero-k.info/Battles/Detail/435867

**Comment by GoogleFrog** [comment]
If you give an attack command at max range while Starlight is rising then it forgets to keep aiming while rising. The result is this.

**Comment by sprunk** [comment]
Why aren't superweapons global? They're supposed to be game enders but it's possible to build two that are outside each other's range.

**Comment by GoogleFrog** [comment]
DRP is not global due to ballistic projectile physics. It could have its physicall maximum range. There is no limitation on the range for the other two superweapons. This is something to consider. The limited range is relevant on speedmetal and large FFA maps.

Superweapons are not actually supposed to directly end the game.

**Comment by Anarchid** [comment]
Zenith and Starlight shoot in a straight line from a given altitude, which would make them clip terrain even if given infinite weapon range. A practical infinite-range version of them would have to have an infinitely high vantage point. DRP actually does not have this problem.

**Comment by GoogleFrog** [comment]
"Clip terrain" here does not mean that they have any bug with clipping. I would say that at long ranges they are liable to be blocked by terrain. This is entirely intended. As I said, they are not necessarily meant to end games.

I once held out against a starlight for quite a while in a 4 way FFA on energy spire plain with emp missiles and a little terraforming to stop my units walking through the beam. At the time an emped starlight did not have its beam disabled.

I've also played a 3 way FFA (originally 6 way) where all players had at least one Zenith and a lot of shields. We lacked other superweapons, or ways to deal with shields, since we were all Core and the arm constructor wreckages had since been destroyed.
